File: kimbyulyi/PGS/Level1/가장가까운같은글자.py
def solution(s):
    answer = [0] * len(s)
    for i in range(len(s)):
        tmp = s[0:i]
        if s[i] in tmp:
            answer[i] = i - tmp.rfind(s[i])
            print(answer[i])
        else:
            answer[i] = -1

    return answer

s = "banana"
print(solution(s))

File: baekyeonsu/가장 가까운 같은 글자.js
function solution(s) {
  let answer = new Array(s.length).fill(-1);
  for (let i = 1; i < s.length; i++) {
    for (let j = i - 1; j >= 0; j--) {
      if (s[j] === s[i]) {
        answer[i] = i - j;
        break;
      }
    }
  }
  return answer;
}
